### PR TITLE
Add zephyr,display chosen property and use it to obtain default display controller in the samples

### PR DESCRIPTION
**Pull request by jfischer-no**
This allows us to make samples that use the display (controller) API more generic and remove ifdefery like
`#define DISPLAY_DEV_NAME DT_LABEL(DT_INST(0, ilitek_ili9340))`.

This patch adds bindings for the display controller integrated in a display, like SSD1306.
We also have devices in the tree where panel and controller are separate entities and
are connected via a defined interface, see mimxrt10xx boards. These are described
differently in devicetree. It may make sense to unify it even if it seems a bit to excessive
for device like SSD1306.
However it is about how the drivers are configured, application that uses display (controller)
API is not affected much since the API provides an interface to retrieve the runtime information,
except it is LVGL adaption layer. It is not possible completely remove options like
LVGL_DISPLAY_DEV_NAME without heavy reworking of LVGL adaption layer
(or complete removal of LVGL from zephyr tree, for which I immediately give +1),
because lvgl_init needs the name of the device, and LVGL_*_RES_MAX options to define
internal buffers.

### CODE REVIEW

**Review by gmarull — APPROVED** [review]
Thanks for these changes, nice cleanup

**Review by MaureenHelm — APPROVED** [review]

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
This needs to be documented in the list of chosen nodes

File: dts/bindings/display/ilitek,ili9xxx-common.yaml
Comment (mbolivar-nordic) [inline dts/bindings/display/ilitek,ili9xxx-common.yaml:7]: This seems to break bisectability by adding required `height` and `width` properties to the following bindings without updating in-tree users:

- dts/bindings/display/ilitek,ili9340.yaml
- dts/bindings/display/ilitek,ili9488.yaml

The dts/bindings/display/ilitek,ili9341.yaml binding is also affected, but doesn't seem to have any in-tree users.

File: boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts
Comment (mbolivar-nordic) [inline boards/arm/bl5340_dvk/bl5340_dvk_cpuapp_common.dts:208]: I think these sorts of changes should be put into the previous patch which added the properties to the binding, for bisectability.

File: samples/drivers/display/src/main.c
Comment (mbolivar-nordic) [inline samples/drivers/display/src/main.c:188]: `display_dev->name`. Here and elsewhere in this patch, getting the name from the device struct seems a little cleaner.

File: boards/shields/ls0xx_generic/Kconfig.defconfig
Comment (mbolivar-nordic) [inline boards/shields/ls0xx_generic/Kconfig.defconfig:18]: Do you also need a separate patch removing these? These values should be coming from devicetree now, right?

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
Thanks very much for this cleanup. I've got a few more comments now that I had time to do a more detailed review; my earlier -1 was just to keep this from being merged too quickly.

File: drivers/display/ssd16xx.c
Comment (jfischer-no) [reply drivers/display/ssd16xx.c:640]: not just moving around, `= dev->config;` vs. `= driver->config;`
it is part of https://github.com/zephyrproject-rtos/zephyr/pull/39708

File: dts/bindings/display/ilitek,ili9xxx-common.yaml
Comment (jfischer-no) [reply dts/bindings/display/ilitek,ili9xxx-common.yaml:7]: yes, realized it later, I will do better.

File: boards/shields/ls0xx_generic/Kconfig.defconfig
Comment (jfischer-no) [reply boards/shields/ls0xx_generic/Kconfig.defconfig:18]: Yes, except for mimxrt10xx boards.

**Review by vanwinkeljan — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]
LGTM pending Marti's requested changes.

**Review by brockus-zephyr — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

File: drivers/display/display_dummy.c
Comment (jfischer-no) [reply drivers/display/display_dummy.c]: I was lazy, reworked both, display_dummy.c and display_sdl.c, to support multi-instance.

File: drivers/display/ssd16xx.c
Comment (mbolivar-nordic) [reply drivers/display/ssd16xx.c:640]: Ah, thanks! I missed that.

File: samples/subsys/display/cfb_custom_font/src/main.c
Comment (mbolivar-nordic) [inline samples/subsys/display/cfb_custom_font/src/main.c:23]: This appears to be an unrelated change for this commit; what's going on here?

Comment (jfischer-no) [reply samples/subsys/display/cfb_custom_font/src/main.c:23]: sloppiness, will update after https://github.com/zephyrproject-rtos/zephyr/pull/40876 is merged.

Comment (mbolivar-nordic) [reply samples/subsys/display/cfb_custom_font/src/main.c:23]: #40876 doesn't seem to have anything to do with this; is that the right link?

Comment (jfischer-no) [reply samples/subsys/display/cfb_custom_font/src/main.c:23]: yes, if I push now, CI will fail again because of that.

**Comment by mbolivar-nordic** [comment]
` samples: crypto: exclude CTR mode for mbedtls `

Did you mean to include this commit in this PR?

**Comment by jfischer-no** [comment]
> `samples: crypto: exclude CTR mode for mbedtls`
> 
> Did you mean to include this commit in this PR?

rather to test what happens,
https://github.com/zephyrproject-rtos/zephyr/pull/40765/checks?check_run_id=4447344836
most of the time I can fix the bugs that CI uncovers myself, but here actions log offers no help at all.

**Comment by mbolivar-nordic** [comment]
> rather to test what happens,

Not really sure if that's a yes or a no :). But  #40876 is merged now, so I'm guessing you plan on making one more update to this PR before we can merge it, right?

**Comment by jfischer-no** [comment]
> > rather to test what happens,
> 
> Not really sure if that's a yes or a no :). But #40876 is merged now, so I'm guessing you plan on making one more update to this PR before we can merge it, right?

Actually, nothing more is planned.  What annoys me is only the unstable CI.

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
> nothing more is planned

OK, I see now you did split the pixel format change into its own commit, but you just didn't mention it. LGTM, thanks.

**Comment by jfischer-no** [comment]
rebased to include 474d6866d02d5b74efe9e991c9ed7db5f436ca5b

**Pull request merged**
